test-data/unit/check-literal.test

```diff
@@ -2960,3 +2960,90 @@ class C(B[Literal["word"]]):
 reveal_type(C().collection)  # N: Revealed type is "builtins.list[Literal['word']]"
 reveal_type(C().word)  # N: Revealed type is "Literal['word']"
 [builtins fixtures/tuple.pyi]
+
+[case testLiteralAddition]
+from typing import Union
+from typing_extensions import Literal
+
+str_a: Literal["a"]
+str_b: Literal["b"]
+str_union_1: Literal["a", "b"]
+str_union_2: Literal["d", "c"]
+s: str
+int_1: Literal[1]
+int_2: Literal[2]
+int_union_1: Literal[1, 2]
+int_union_2: Literal[4, 3]
+i: int
+bytes_a: Literal[b"a"]
+bytes_b: Literal[b"b"]
+bytes_union_1: Literal[b"a", b"b"]
+bytes_union_2: Literal[b"d", b"c"]
```

Comment:
Test union with literal type and non-literal type -- e.g. user-defined class that defines `__add__` that accepts `str`. Also test with union of literal type and `Any`.

Reply:
The narrowing is only applied if all Union items are Literals itself. Thus the inference for these cases didn't change.

Test cases added in https://github.com/python/mypy/pull/18004/commits/36f5e2ead9b02bd6922f88b1761fb70470978011